pkg/api/orchestrators.go

```diff
@@ -157,59 +157,48 @@ func kubernetesInfo(csOrch *OrchestratorProfile) ([]*OrchestratorVersionProfile,
 
 func kubernetesUpgrades(csOrch *OrchestratorProfile) ([]*OrchestratorProfile, error) {
 	ret := []*OrchestratorProfile{}
-	var err error
+
+	currentVer, err := semver.NewVersion(csOrch.OrchestratorVersion)
+	if err != nil {
+		return nil, err
+	}
+	currentMajor, currentMinor, currentPatch := currentVer.Major(), currentVer.Minor(), currentVer.Patch()
+	var nextMajor, nextMinor int64
 
 	switch {
-	case strings.HasPrefix(csOrch.OrchestratorVersion, "1.5"):
-		// add next version
-		ret = append(ret, &OrchestratorProfile{
-			OrchestratorType:    Kubernetes,
-			OrchestratorVersion: common.KubernetesVersion1Dot6Dot11,
-		})
-	case strings.HasPrefix(csOrch.OrchestratorVersion, "1.6"):
-		// check for patch upgrade
-		if ret, err = addPatchUpgrade(ret, csOrch.OrchestratorVersion, common.KubernetesVersion1Dot6Dot11); err != nil {
-			return ret, err
+	case currentMajor == 1 && currentMinor == 5:
+		nextMajor = 1
+		nextMinor = 6
+	case currentMajor == 1 && currentMinor == 6:
+		nextMajor = 1
+		nextMinor = 7
+	case currentMajor == 1 && currentMinor == 7:
+		nextMajor = 1
+		nextMinor = 8
+	}
+	for _, ver := range common.AllKubernetesSupportedVersions {
+		nextVersion, err := semver.NewVersion(ver)
+		if err != nil {
+			continue
 		}
-		// add next version
-		ret = append(ret, &OrchestratorProfile{
-			OrchestratorType:    Kubernetes,
-			OrchestratorVersion: common.KubernetesVersion1Dot7Dot9,
-		})
-	case strings.HasPrefix(csOrch.OrchestratorVersion, "1.7"):
-		// check for patch upgrade
-		if ret, err = addPatchUpgrade(ret, csOrch.OrchestratorVersion, common.KubernetesVersion1Dot7Dot9); err != nil {
-			return ret, err
+		// add patch upgrade
+		if nextVersion.Major() == currentMajor && nextVersion.Minor() == currentMinor && currentPatch < nextVersion.Patch() {
+			ret = append(ret, &OrchestratorProfile{
+				OrchestratorType:    Kubernetes,
+				OrchestratorVersion: ver,
+			})
 		}
 		// add next version
-		ret = append(ret, &OrchestratorProfile{
-			OrchestratorType:    Kubernetes,
-			OrchestratorVersion: common.KubernetesVersion1Dot8Dot2,
-		})
-	case strings.HasPrefix(csOrch.OrchestratorVersion, "1.8"):
-		// check for patch upgrade
-		if ret, err = addPatchUpgrade(ret, csOrch.OrchestratorVersion, common.KubernetesVersion1Dot8Dot2); err != nil {
-			return ret, err
+		if nextVersion.Major() == nextMajor && nextVersion.Minor() == nextMinor {
```

Comment:
I don't like how this is custom logic that can get out of sync from our list of all orcherstators. Why not just get the list of all orcherstators/versions and pick the one we are on? That way they never get out of sync.